diff -Nur bochs-2.0.2.orig/configure.in bochs-2.0.2.chng/configure.in
--- bochs-2.0.2.orig/configure.in	Wed Jan 22 14:18:28 2003
+++ bochs-2.0.2.chng/configure.in	Wed Sep 10 13:37:11 2003
@@ -1660,7 +1660,7 @@
 # check for existence of and usable version of wxWindows.  This comes
 # early so that it could affect which one gets chosen.
 if test x$WX_CONFIG = x; then
-  AC_CHECK_PROG(WX_CONFIG, [ wx-config --version ], wx-config, not_found)
+  AC_CHECK_PROG(WX_CONFIG, [ wxgtk-2.4-config --version ], wxgtk-2.4-config, not_found)
 fi
 AC_MSG_CHECKING(for wxWindows configuration script)
 AC_MSG_RESULT($WX_CONFIG)
